### PR TITLE
export tests: patch instead of updates where it makes sense

### PR DESCRIPTION
**Pull request by akalenyu**
### What this PR does
- use patch to start/stop VM
there's no reason to loop over an update with eventually
when there's an API that uses patch
- use patch for kv cert config changes
Although unlikely, the kubevirt install may change and thus cause the update calls to fail.
We can instead specifically issue a patch against the fields we care
about.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: tests/storage/export.go [hunk 1]
```diff
@@ -1284,7 +1255,10 @@ var _ = SIGDescribe("Export", func() {
 			vm.Spec.RunStrategy = virtpointer.P(v1.RunStrategyAlways)
 		}
 		vm = createVM(vm)
-		stopVM(vm)
+		if libstorage.IsStorageClassBindingModeWaitForFirstConsumer(sc) {
+			err = virtClient.VirtualMachine(vm.Namespace).Stop(context.Background(), vm.Name, &v1.StopOptions{})
+			Expect(err).ToNot(HaveOccurred())
+		}
```
Comment (orelmisan) [inline tests/storage/export.go:1261]: AFAIU from the existing logic, this condition is new.
Could you please add reasoning in the commit message explaining why was it added?

**Review by orelmisan — COMMENTED** [review]
Thank you for the PR @akalenyu.

File: tests/storage/export.go [hunk 2]
```diff
@@ -2015,14 +1999,18 @@ var _ = SIGDescribe("Export", func() {
 			if equality.Semantic.DeepEqual(beforeCertParams, &kv.Spec.CertificateRotationStrategy) {
 				return
 			}
-			kv.Spec.CertificateRotationStrategy = *beforeCertParams
-			_, err := virtClient.KubeVirt(kv.Namespace).Update(context.Background(), kv, metav1.UpdateOptions{})
+			patchSet := patch.New(
+				patch.WithAdd("/spec/certificateRotateStrategy", beforeCertParams),
+			)
+			patchBytes, err := patchSet.GeneratePayload()
```
Comment (alicefr) [inline tests/storage/export.go:2005]: nit
```suggestion
			patchBytes, err := patch.New(
				patch.WithAdd("/spec/certificateRotateStrategy", beforeCertParams),
			).GeneratePayload()
```

**Comment by alicefr** [comment]
Looks good!
/approve
/hold
To address the comments

File: tests/storage/export.go (hunk 2 above)
Comment (akalenyu) [reply tests/storage/export.go]: done

File: tests/storage/export.go (hunk 1 above)
Comment (akalenyu) [reply tests/storage/export.go:1261]: sure, added

**Review by ShellyKa13 — APPROVED** [review]
/lgtm

File: tests/storage/export.go (hunk 1 above)
Comment (orelmisan) [reply tests/storage/export.go:1261]: Could you please elaborate how "avoid stopping when the VM wasn't started" relates to the newly added condition (which AFAIU relates to a storageClass)?

Comment (akalenyu) [reply tests/storage/export.go:1261]: The same condition is used to decide whether we start the VM a few lines above

Comment (orelmisan) [reply tests/storage/export.go:1261]: I see, thank you for the explanation.
Please consider adding in follow-up, a flag to unify the two conditions.

**Pull request merged**